src/main/java/com/rocketden/main/service/SubmitService.java

```diff
@@ -36,15 +42,52 @@ public SubmissionDto submitSolution(Game game, SubmissionRequest request) {
         player.setPlayerCode(playerCode);
 
         // Create a dummy submission - this will be replaced with a call to the tester
+        List<Problem> problems = game.getProblems();
+        int numTestCases = problems.isEmpty() ? 10 : problems.get(0).getTestCases().size();
+
         Submission submission = new Submission();
         submission.setPlayerCode(playerCode);
-        submission.setNumCorrect(10);
-        submission.setNumTestCases(10);
+        submission.setNumCorrect(numTestCases);
+        submission.setNumTestCases(numTestCases);
 
         player.getSubmissions().add(submission);
         player.setSolved(true);
 
+        // Sort list of players by who is winning
+        GameDto gameDto = GameMapper.toDto(game);
+        sortLeaderboard(gameDto.getPlayers());
+
+        // Send socket update with latest leaderboard info
+        // TODO: test socket in GameSocketTests when merged
```

Comment:
Yeah that was my plan since the GameSocketTests file hasn't actually been merged into master yet - it looks like that PR has been approved tho so I could also merge that PR and pull the file from master and implement it in this branch too - lmk

Reply:
If that's doable, I think that'd be great! The less TODOs in the PR, the better.

Reply:
Sounds good, I'll do that - also, since that knocks one of the items off the future changes list, I'll also go ahead and do the "move sort leaderboard to game mapper" feature in this PR too then (since the remaining changes would be small enough where it doesn't warrant a separate PR) - it means this PR will again be slightly bigger haha but hopefully it's not too much more (since it's just moving the logic to a different file)

Note: moving the sorting to game mapper is necessary because as it stands only the submit endpoint will sort the leaderboard before sending the socket update - moving the sorting to the toDto function will ensure all of them have the correct leaderboard (which I only realized once I already finished coding it)

Reply:
This all makes sense -- thanks for adding this!